main/src/ca/uwaterloo/flix/language/ast/ResolvedAst.scala

```diff
@@ -105,6 +105,8 @@ object ResolvedAst {
 
     case class Apply(exp: Expr, exps: List[Expr], loc: SourceLocation) extends Expr
 
+    case class ApplyDef(exp: Expr.Def, exps: List[Expr], loc: SourceLocation) extends Expr
```

Comment:
This is not right. You should not be referring to Expr.Def. Instead, you should always refer to the relevant symbol.

Reply:
We generally never have ADTs that refer to specific subcases of (other) ADTs -- because it creates a tight coupling.

The point of ApplyDef is that it targets a Def, and a Def has a DefnSym

Reply:
Roger that

Reply:
I think the important thing here was to make this compile and run with minimal changes (everything is in https://github.com/flix/flix/pull/8667 )

Reply:
Agree on minimalism. But its about having the right abstraction. In this PR you have this weird ApplyDef is not its own thing but a reference to another thing, yet it still has some extra type vars etc.